action/protocol/staking/vote_reviser.go

```diff
@@ -47,9 +47,11 @@ func (vr *VoteReviser) Revise(ctx protocol.FeatureCtx, csm CandidateStateManager
 		case err != nil:
 			return err
 		}
-		cands, err = vr.reviseSelfStakeBuckets(ctx, csm, height, cands)
-		if err != nil {
-			return err
+		if vr.shouldReviseSelfStakeBuckets(height) {
```

Comment:
it is actually revise for endorsement status, i think name it `reviseEndorsement` or at least put `endorsement` into the func name